plugins/inventory/vmware_vm_inventory.py

```diff
@@ -314,41 +335,112 @@ def check_requirements(self):
             raise AnsibleError("Missing one of the following : hostname, username, password. Please read "
                                "the documentation for more information.")
 
-    @staticmethod
-    def _process_object_types(vobj, level=0):
```

Comment:
This change is required for handling JSON type serialization. See https://github.com/ansible-collections/vmware/pull/58

Reply:
The PR have been tested successfully with every possible properties ( `properties: ALL` )  including YAML and TOML. Thus, overriding the changes in  #58 .  Unless there is a test case I missed.

cc:  @AlanCoding